jina/executors/indexers/keyvalue.py

```diff
@@ -50,29 +63,39 @@ class _ReadHandler:
     """
 
     def __init__(self, path, key_length):
-        with open(path + '.head', 'rb') as fp:
-            tmp = np.frombuffer(
-                fp.read(),
-                dtype=[
-                    ('', (np.str_, key_length)),
-                    ('', np.int64),
-                    ('', np.int64),
-                    ('', np.int64),
-                ],
-            )
-            self.header = {
-                r[0]: None
-                if np.array_equal((r[1], r[2], r[3]), HEADER_NONE_ENTRY)
-                else (r[1], r[2], r[3])
-                for r in tmp
-            }
-        self._body = open(path, 'r+b')
+        self.path = path
+        self.header = {}
+        if os.path.exists(self.path + '.head'):
```

Comment:
add an `else` with a raise

Reply:
no raise, this happens when there is no data in the shard. This was hidden because before file was always created just bu creating the WriteHandler